opslevel/resource_opslevel_team.go

```diff
@@ -149,30 +143,33 @@ func (teamResource *TeamResource) Create(ctx context.Context, req resource.Creat
 	if len(members) > 0 {
 		teamCreateInput.Members = &members
 	}
-
 	if planModel.Parent.ValueString() != "" {
 		teamCreateInput.ParentTeam = opslevel.NewIdentifier(planModel.Parent.ValueString())
 	}
+
 	team, err := teamResource.client.CreateTeam(teamCreateInput)
 	if err != nil || team == nil {
 		resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("unable to create team, got error: %s", err))
 		return
 	}
-	err = team.Hydrate(teamResource.client)
-	if err != nil {
-		resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("unable to hydrate team, got error: %s", err))
-		return
-	}
 
-	aliases, diags := SetValueToStringSlice(ctx, planModel.Aliases)
-	resp.Diagnostics.Append(diags...)
-	if err = teamResource.reconcileTeamAliases(team, aliases); err != nil {
-		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("unable to reconcile aliases, got error: %s", err))
-		return
+	if len(planModel.Aliases.Elements()) > 0 {
+		aliases, diags := SetValueToStringSlice(ctx, planModel.Aliases)
+		if diags != nil && diags.HasError() {
+			resp.Diagnostics.AddError("Config error", fmt.Sprintf("Unable to handle given team aliases: '%s'", planModel.Aliases))
+			return
+		}
+		if err = team.ReconcileAliases(teamResource.client, aliases); err != nil {
+			resp.Diagnostics.AddWarning("opslevel client error", fmt.Sprintf("warning while reconciling team aliases: '%s'\n%s", aliases, err))
+			resp.Diagnostics.AddWarning("Config warning", "On create, OpsLevel API creates a new alias for teams. If this causes issues, create team with empty 'aliases'. Then update team with 'aliases'")
+		}
 	}
 
 	createdTeamResourceModel, diags := NewTeamResourceModel(ctx, *team, planModel)
 	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
```

Comment:
this is logically different then `if diags.HasError()` are we sure we want to bail here if there has been any error added up until this point and not just if `NewTeamResourceModel` had errors?

Reply:
🤔 In this function, every `resp.Diagnostics.AddError(...)` is followed by a `return` and there is no `resp.Diagnostics.Append(diags...)` before the line above. Each `if diags.HasError() { ... }` contains a `return`. I'm not seeing the case where an error was added to `resp.Diagnostics` but not returned

Reply:
then why `resp.Diagnostics.Append(diags...)` ?

Reply:
oh because warnings 🤦